duesseldorf.py

```diff
@@ -0,0 +1,163 @@
+
+    days = ('Montag', 'Dienstag', 'Mittwoch', 'Donnerstag', 'Freitag')
+    for day in days:
+        data = document.find('div', {'data-day': day})
+        date = data.attrs['data-date']
+        parse_day(canteen, date, data)
+
+    return canteen.toXMLFeed()
+
+
+def parse_day(canteen, date, data):
+    # get category names from headings
+    counter_names = data.find_all('h2')
+    # get meals from each category
+    counters = data.find_all('div', class_='counter-table')
+    add_meals_from_counter(canteen, counter_names, counters, date)
+
+
+def add_meals_from_counter(canteen, counter_names, counters, date):
+    i = 0
+    for counter in counters:
+        # get correct category name
+        counter_name = counter_names[i].text.strip()
+        # find all menues for this category
+        menu = counter.find('ul', 'menu')
+        if menu:
+            # get contents of menu
+            items = menu.find_all('li')
+            # parse menu items
+            meal, raw_legend = parse_menu_items(items)
+            # Studentenwerk Düsseldorf does not maintain a easily readable list of additives etc.
+            # need to do our own parsing against a static list
+            legend = parse_legend(raw_legend)
+            # get prices for each role
+            prices = parse_prices(counter.find('ul', 'price'))
+            # add meal to list
+            canteen.addMeal(date, counter_name, meal, notes=legend, prices=prices)
+        i += 1
+
+
+def parse_legend(raw_legend):
+    verbose_legend = []  # we'll add parsed legend entries here
+    for li in raw_legend:  # raw_legends contains a two dimensional array
+        for li2 in li:  # second dimension
+            for key in legend_dict:  # iterate over keys and check against legend item (li2)
+                if li2 == key:
+                    # if found, add verbose description to list and cancel current iteration
+                    verbose_legend.append(legend_dict[key])
+                    break
+    return verbose_legend
+
+
+def parse_menu_items(items):
+    strings = []
+    raw_legend = []  # this will just contain lists of legend keys
+    for item in items:
+        # collect text from current item
+        text = item.text.strip()
+        if text not in ['', 'Mehr Informationen']:
+            # find notes in text
+            legend_tmp = legend_regex.findall(text)
+            for li in legend_tmp:
+                # remove notes from text to obtain a clean string
+                text = text.replace(li, '').strip()
+                # in addition, remove brackets left over after note deletion
+                for ch in ['(', ')', '[', ']']:
+                    li = li.replace(ch, '')
+                # split notes in list and append master list
+                li = li.split(',')
+                raw_legend.append(li)
+            strings.append(text)
+    # join all strings and return result + raw legend
+    return (", ".join(strings), raw_legend)
```

Comment:
Agreed for "- Preis ab:". As for "- Preis je 100g:" I'm not sure if it's not better to keep this one as the shown prices are really only per 100 g of salad (you'll have to weight it before checkout)